public/app/features/explore/ContentOutline/ContentOutline.tsx

```diff
@@ -25,27 +42,52 @@ const getStyles = (theme: GrafanaTheme2) => {
       top: 0,
     }),
     buttonStyles: css({
-      textAlign: 'left',
+      textAlign: 'right',
       width: '100%',
       padding: theme.spacing(0, 1.5),
+      '&:hover': {
+        color: theme.colors.text.primary,
+        textDecoration: 'underline',
+      },
+    }),
+    iconButton: css({
+      justifyContent: 'center',
+      width: theme.spacing(4),
+      '&:hover': {
+        color: theme.colors.text.primary,
+        background: theme.colors.background.secondary,
+        textDecoration: 'underline',
+      },
+    }),
+    sectionWrapper: css({
+      display: 'flex',
+    }),
+    indentRoot: css(baseIndentStyle),
+    indentChildren: css({
+      ...baseIndentStyle,
+      marginLeft: expanded ? INDENT_LEVELS.CHILD_EXPANDED : INDENT_LEVELS.CHILD_COLLAPSED, // override the marginLeft property
     }),
   };
 };
 
 export function ContentOutline({ scroller, panelId }: { scroller: HTMLElement | undefined; panelId: string }) {
+  const [contentOutlineExpanded, toggleContentOutlineExpanded] = useToggle(false);
+  const [sectionExpanded, toggleSectionExpanded] = useToggle(false);
+  const styles = useStyles2((theme) => getStyles(theme, contentOutlineExpanded));
   const { outlineItems } = useContentOutlineContext();
-  const [expanded, toggleExpanded] = useToggle(false);
   const [activeItemId, setActiveItemId] = useState<string | undefined>(outlineItems[0]?.id);
-  const styles = useStyles2((theme) => getStyles(theme));
   const scrollerRef = useRef(scroller || null);
   const { y: verticalScroll } = useScroll(scrollerRef);
 
-  const scrollIntoView = (ref: HTMLElement | null, buttonTitle: string) => {
+  const scrollIntoView = (ref: HTMLElement | null, itemPanelId: string) => {
     let scrollValue = 0;
     let el: HTMLElement | null | undefined = ref;
 
+    // This is to handle ContentOutlineItem wrapping each QueryEditorRow
+    const customOffsetTop = itemPanelId === 'Queries' ? QUERY_ROW_TOP_OFFSET : 0;
```

Comment:
Why exactly do we need it? Would be nice not to have it all :D in worst case I think it should be provided when being registered so the Content Outline doesn't have dependency on specific usage. We may want to externalize this component in the future.

Reply:
This is when users want to get to a specific query row inside the queries container. The difference is this:
Without offset
![Screenshot 2024-01-19 at 11 47 40 AM](https://github.com/grafana/grafana/assets/58232930/b1de03b3-4a4f-472f-aa2d-78a212f21bf3)
With offset:
![Screenshot 2024-01-19 at 11 47 53 AM](https://github.com/grafana/grafana/assets/58232930/abf8a7b0-0205-4ac5-be7f-141b7d04765a)

Having the offset adds a little bit of margin from the top so the query row is fully visible